tests/external/rule_conditions_test.php

```diff
@@ -0,0 +1,155 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_cohortmanager\external;
+
+use externallib_advanced_testcase;
+use tool_cohortmanager\condition;
+use tool_cohortmanager\rule;
+use tool_cohortmanager\tool_cohortmanager\condition\cohort_membership;
+use tool_cohortmanager\tool_cohortmanager\condition\user_profile;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/lib/externallib.php');
+require_once($CFG->dirroot . '/webservice/tests/helpers.php');
+
+/**
+ * Tests for matching users external APIs .
+ *
+ * @package    tool_cohortmanager
+ * @author     Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class rule_conditions_test extends externallib_advanced_testcase {
+
+    /**
+     * Test exception if rule is not exist.
```

Comment:
Should `@cover` tag be specified?
```suggestion
     * Test exception if rule is not exist.
     * @covers ::get_conditions
```